src/main/java/org/apache/sysml/hops/rewrite/RewriteConstantFolding.java

```diff
@@ -131,7 +131,11 @@ else if( isApplicableTrueDisjunctivePredicate(root) )
 		{
 			literal = new LiteralOp(true);
 		}
-		
+		//fold biconditional predicate if both the inputs are same
+		else if(isApplicableFalseBiconditionalPredicate(root) )
```

Comment:
the name `biconditional predicate` may or may not be appropriate. But, the logic seems correct.